### PR TITLE
chore: Merge main into v2

### PR DESCRIPTION
**Pull request by scottgerring**
**Issue #, if available:**
Periodic merge of main changes down into V2. This should be merged and not squashed to make life easier the next time.

## Description of changes:

**Checklist**

 

* [x] [Meet tenets criteria](https://docs.powertools.aws.dev/lambda-java/#tenets)
* [ ] Update tests
* [ ] Update docs
* [ ] PR title follows [conventional commit semantics](https://www.conventionalcommits.org/en/v1.0.0/)

## Breaking change checklist

**RFC issue #**:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by scottgerring** [comment]
This should not be squashed!

File: examples/powertools-examples-batch/pom.xml [hunk 1]
```diff
@@ -14,8 +14,9 @@
         <log4j.version>2.20.0</log4j.version>
         <maven.compiler.source>1.8</maven.compiler.source>
         <maven.compiler.target>1.8</maven.compiler.target>
-        <sdk.version>2.20.133</sdk.version>
+        <sdk.version>2.20.152</sdk.version>
```
Comment (jeromevdl) [inline examples/powertools-examples-batch/pom.xml:17]:
```suggestion
        <sdk.version>2.20.162</sdk.version>
```

File: examples/powertools-examples-core-utilities/cdk/app/pom.xml [hunk 2]
```diff
@@ -67,80 +68,72 @@
...
-                <groupId>org.apache.maven.plugins</groupId>
-                <artifactId>maven-deploy-plugin</artifactId>
-                <configuration>
-                    <skip>true</skip>
-                </configuration>
-            </plugin>
-        </plugins>
+      <finalName>helloworld-lambda</finalName>
+      <plugins>
+        <plugin>
```
Comment (jeromevdl) [inline examples/powertools-examples-core-utilities/cdk/app/pom.xml:73]: not the good indentation... was the other one...

File: pom.xml [hunk 3]
```diff
@@ -74,19 +74,19 @@
         <log4j.version>2.20.0</log4j.version>
         <jackson.version>2.15.2</jackson.version>
         <aspectj.version>1.9.7</aspectj.version>
-        <aws.sdk.version>2.20.136</aws.sdk.version>
+        <aws.sdk.version>2.20.162</aws.sdk.version>
         <aws.xray.recorder.version>2.14.0</aws.xray.recorder.version>
         <payloadoffloading-common.version>2.1.3</payloadoffloading-common.version>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <lambda.core.version>1.2.3</lambda.core.version>
+        <lambda.core.version>1.2.2</lambda.core.version>
```
Comment (jeromevdl) [inline pom.xml:81]:
```suggestion
        <lambda.core.version>1.2.3</lambda.core.version>
```

**Review by jeromevdl — COMMENTED** [review]
just a few things to fix

**Pull request merged**